### ISSUE TITLE
unused_unit false positive for macro inputs

### ISSUE THREAD

**Issue by goffrie** [rust-lang/rust-clippy#4076]
sample test case:
```
use crossbeam_channel::{unbounded, select};
fn main() {
    let (tx,rx) = unbounded();
    tx.send("hi").unwrap();
    select! {
        recv(rx) -> _x => (),
    }
}
```
```
% cargo clippy
    Checking asdf v0.1.0 (/tmp/asdf)
warning: unneeded unit expression
 --> src/main.rs:6:27
  |
6 |         recv(rx) -> _x => (),
  |                           ^^ help: remove the final `()`
  |
  = note: #[warn(clippy::unused_unit)] on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unused_unit

    Finished dev [unoptimized + debuginfo] target(s) in 0.24s
```

of course, following the suggestion leads to invalid code.

`unused_unit` shouldn't fire for unit expressions that came from a macro argument.

**Comment by phansch** [comment]
Minimal repro:

```rust
#![warn(clippy::unused_unit)]

fn main() {
    macro_rules! foo {
        (recv($r:expr) -> $res:pat => $body:expr) => {
            $body
        }
    }

    foo! {
        recv(rx) -> _x => ()
    }
}
```
